controllers/managedfusionoffering_controller.go

```diff
@@ -0,0 +1,82 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controller "sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// ManagedFusionOfferingReconciler reconciles a ManagedFusionOffering object
+type ManagedFusionOfferingReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	ctx                   context.Context
+	managedFusionOffering *v1alpha1.ManagedFusionOffering
+}
+
+//+kubebuilder:rbac:groups=misf.ibm.com,resources=managedfusionofferings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=misf.ibm.com,resources=managedfusionofferings/status,verbs=get;update;patch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ManagedFusionOfferingReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *controller.Options) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.ManagedFusionOffering{}).
+		Complete(r)
+}
+
+func (r *ManagedFusionOfferingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("req.Namespace", req.Namespace, "req.Name", req.Name)
+	log.Info("Starting reconcile for ManangedFusionOffering")
+
+	r.initReconciler(ctx, req)
+
+	if err := r.get(r.managedFusionOffering); err != nil {
+		return ctrl.Result{}, fmt.Errorf("Unable to get managedFusionOffering: %v", err)
+	}
+
+	switch r.managedFusionOffering.Spec.Kind {
+	case v1alpha1.KindDataFoundation:
+	case v1alpha1.KindDataFoundationClient:
+	default:
+		return ctrl.Result{}, fmt.Errorf("Unknown managedFusionOffering Kind: %v", r.managedFusionOffering.Spec.Kind)
+	}
```

Comment:
This check and the enum itself do not make sense.
The plan is to introduce offerings as plugins, so any value should be ok as long as we can find a plugin for it and I don't see how can we do it in compile time. as such we should assume the field a a plain old string and no validation for now

Reply:
```suggestion
	if result, err := r.reconcileDF(); err != nil {
		r.Log.Error(err, "An error was encountered during reconcile of DF offering")
		return ctrl.Result{}, err
	} else {
		return result, nil
	}
```

So for now(without a plugin system), can I assume that a function called reconcileDF function will be called in Reconcile function?

Reply:
First, For this PR nothing should be called as this is not a DF PR.
Second, we already discussed in the morning should abstract it behind a `pluginReconcile(...)` call the internally calls your `reconcileDF`. Of course, `pluginReconcile` is not a method of the reconciler so you will need to pass the reconciler as a parameter. Something like:

```
result, err := pluginReconcile(r, ...);  
```